restfull-api/routes/todos.js

```diff
@@ -1,4 +1,72 @@
 const express = require("express");
 const router = express.Router();
+const Todo = require("../models/todo");
+const todo = require("../models/todo");
+
+//getting all
+router.get("/", async (req, res) => {
+  try {
+    const todos = await Todo.find();
+    res.json(todos);
+  } catch (err) {
+    res.status(500).json({ message: err.message });
+  }
+});
+//getting one
+router.get("/:id", getTodo, (req, res) => {
+  res.json(res.todo);
+});
```

Comment:
```suggestion
router.get("/:id", getTodo, (req, res) => {
  try {
    todo = await Todo.findById(req.params.id);
    if (!todo) {
      res.status(404).json({})
    }
    res.json(todo)
  } catch(error) {
        res.status(500).json({})
  }
});
```

Reply:
i think i did not get the point

Reply:
The point is that you write the logic for finding the todo from the DB here instead of using the `getTodo` middleware.